### PR TITLE
Issue #850: Clarify XA lifecycle

### PR DESCRIPTION
**Pull request by ljacomet**
Changes as proposed in #850

### CODE REVIEW

File: transactions/src/main/java/org/ehcache/transactions/xa/txmgr/provider/FromConfigurationTransactionManagerProviderConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+package org.ehcache.transactions.xa.txmgr.provider;
+
+import org.ehcache.spi.service.ServiceCreationConfiguration;
+import org.ehcache.transactions.xa.txmgr.TransactionManagerWrapper;
+
+/**
+ * {@link ServiceCreationConfiguration} for the
+ * {@link FromConfigurationTransactionManagerProvider}.
+ */
+public interface FromConfigurationTransactionManagerProviderConfiguration extends ServiceCreationConfiguration<TransactionManagerProvider> {
```
Comment (ljacomet) [inline transactions/src/main/java/org/ehcache/transactions/xa/txmgr/provider/FromConfigurationTransactionManagerProviderConfiguration.java]: Let's be honest, this is not my favourite class name. Suggestions for change more than welcome 😉

File: docs/src/docs/asciidoc/user/xa.adoc [hunk 2]
```diff
@@ -30,14 +30,13 @@ supporting the whole two-phase commit protocol, including crash recovery.
...
-The simplest possible configuration is to configure a cache as XA-transactional and let Ehcache 3 figure out everything
-else, meaning it will auto-detect and integrate with the Bitronix transaction manager.
+The simplest possible configuration is to configure a cache manager as transactional aware by using the provided Bitronix transaction manager integration.
 
 This `INFO` log informs you of the detected transaction manager:
 
 [source]
 ----
-INFO o.e.t.x.c.DefaultTransactionManagerProvider - Using looked up transaction manager : a BitronixTransactionManager with 0 in-flight transaction(s)
+INFO org.ehcache.transactions.xa.txmgr.btm.BitronixTransactionManagerProviderConfiguration - Using looked up transaction manager : a BitronixTransactionManager with 0 in-flight transaction(s)
```
Comment (chrisdennis) [inline docs/src/docs/asciidoc/user/xa.adoc]: Should this be generated by parsing the output of a test... ;)

File: transactions/src/main/java/org/ehcache/transactions/xa/internal/xml/TxCacheServiceConfigurationParser.java [hunk 3]
```diff
@@ -13,8 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.ehcache.transactions.xml;
 
+package org.ehcache.transactions.xa.internal.xml;
+
+import org.ehcache.transactions.xa.internal.xml.TxCacheManagerServiceConfigurationParser;
```
Comment (chrisdennis) [inline transactions/src/main/java/org/ehcache/transactions/xa/internal/xml/TxCacheServiceConfigurationParser.java]: import from same package

File: docs/src/docs/asciidoc/user/xa.adoc (hunk 2 above)
Comment (ljacomet) [inline docs/src/docs/asciidoc/user/xa.adoc]: Let me know when you are done with this asciidoc extension 📟

**Comment by ljacomet** [comment]
Major update following conversation with @chrisdennis (resumed on #850 itself) which at least resolves the ugly names part ... and clarifies roles and lifecycle even more I find (but I am biased for sure).

**Pull request merged**